### PR TITLE
Census api david

### PR DESCRIPTION
**Pull request by whymand**
Created "clean_acs" function to pull and clean ACS data; applied to 1 and 5 year ACS data for poverty and educational attainment by race

### CODE REVIEW

File: census_api_function/census_api_function.R [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  pkgs.missing <- pkgs[!check]
+  install.packages(pkgs.missing)
+  check <-
+    sapply(pkgs.missing,
+           require,
+           warn.conflicts = TRUE,
+           character.only = TRUE)
+}
+
+census_api_key("e5ae735f65e6b93f5c2f9bc16bea67b72fa202da")
```
Comment (fansi-sifan) [inline census_api_function/census_api_function.R]: Treat api key as your password - so don't post it to a public repo! Instead, you can [install api keys ](https://walkerke.github.io/tidycensus/reference/census_api_key.html)to your local environment.

File: census_api_function/census_api_function.R [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+# unlike tidycensus's get_acs, clean_acs assigns meaningful variable names
+
+# first argument = geography
+# second argument = variable codes
+# third argument = endyear of survey (yyyy)
+# fourth argument = number of years survey spans (either 1, 3, or 5)
+
+
+
+clean_acs <- function(geography, variables, year, span) {
```
Comment (fansi-sifan) [inline census_api_function/census_api_function.R]: Great job on this one! Super handy to have

File: census_api_function/census_api_function.R [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  
+  
+  return(final)
+  
+}
+
+# VARIABLE CODES =================================================
+
+#variable codes for poverty rate by race (manual lookup & test)
+pov_race_codes<-c("S1701_C03_013",
```
Comment (fansi-sifan) [inline census_api_function/census_api_function.R]: you can use ` purrr::map_chr(seq(13,21),function(x)paste0("S1701_C03_0",x)) ` to 1) simplify script 2) minimize copy-pasting mistakes

**Comment by whymand** [comment]
Thank you for the feedback! I implemented the two changes you recommended.... I see the purrr::map_chr is a very efficient shortcut. I'm also very glad you like my clean_acs function, I certainly learned a lot and had an adventure creating it :)

**Pull request merged**